Test `test_storage_s3_queue/test.py::test_multiple_tables_streaming_sync_distributed` is flaky

**Describe the bug**

https://s3.amazonaws.com/clickhouse-test-reports/0/ccbbe8d0afc71e280a9314dae366d05de166474b/integration_tests__tsan__%5B2_6%5D.html

The test failed 12 times in the part **year** , [the report from CI database.](https://play.clickhouse.com/play?user=play#U0VMRUNUIGNoZWNrX25hbWUsIHRlc3RfbmFtZSwgcmVwb3J0X3VybApGUk9NIGNoZWNrcwpXSEVSRSBjaGVja19uYW1lIExJS0UgJ0ludGVncmF0aW9uJScKICAgIEFORCBjaGVja19zdGFydF90aW1lID49IG5vdygpIC0gSU5URVJWQUwgMzY1IERBWVMKICAgIEFORCBwdWxsX3JlcXVlc3RfbnVtYmVyID0gMAogICAgQU5EIHRlc3Rfc3RhdHVzICE9ICdTS0lQUEVEJwogICAgQU5EIHRlc3Rfc3RhdHVzIExJS0UgJ0YlJwogICAgQU5EIGNoZWNrX3N0YXR1cyAhPSAnc3VjY2VzcycKICAgIEFORCBjaGVja19uYW1lIE5PVCBMSUtFICdTUUxhbmNlciUnCiAgICBBTkQgdGVzdF9uYW1lIElMSUtFICcldGVzdF9tdWx0aXBsZV90YWJsZXNfc3RyZWFtaW5nX3N5bmNfZGlzdHJpYnV0ZWQlJwpPUkRFUiBCWSBjaGVja19uYW1lLCB0ZXN0X25hbWUsIGNoZWNrX3N0YXJ0X3RpbWU=)



**Error message and/or stacktrace**
```
=================================== FAILURES ===================================
___________ test_multiple_tables_streaming_sync_distributed[ordered] ___________
[gw0] linux -- Python 3.10.12 /usr/bin/python3

started_cluster = <helpers.cluster.ClickHouseCluster object at 0x7f2d56ccfa00>
mode = 'ordered'

    @pytest.mark.parametrize("mode", AVAILABLE_MODES)
    def test_multiple_tables_streaming_sync_distributed(started_cluster, mode):
        node = started_cluster.instances["instance"]
        node_2 = started_cluster.instances["instance2"]
        table_name = f"multiple_tables_streaming_sync_distributed_{mode}"
        dst_table_name = f"{table_name}_dst"
        keeper_path = f"/clickhouse/test_{table_name}"
        files_path = f"{table_name}_data"
        files_to_generate = 300
        row_num = 50
        total_rows = row_num * files_to_generate
    
        for instance in [node, node_2]:
            create_table(
                started_cluster,
                instance,
                table_name,
                mode,
                files_path,
                additional_settings={"keeper_path": keeper_path, "s3queue_buckets": 2},
            )
    
        for instance in [node, node_2]:
            create_mv(instance, table_name, dst_table_name)
    
        total_values = generate_random_files(
            started_cluster, files_path, files_to_generate, row_num=row_num
        )
    
        def get_count(node, table_name):
            return int(run_query(node, f"SELECT count() FROM {table_name}"))
    
        for _ in range(150):
            if (
                get_count(node, dst_table_name) + get_count(node_2, dst_table_name)
            ) == total_rows:
                break
            time.sleep(1)
    
        if (
            get_count(node, dst_table_name) + get_count(node_2, dst_table_name)
        ) != total_rows:
            info = node.query(
                f"SELECT * FROM system.s3queue WHERE zookeeper_path like '%{table_name}' ORDER BY file_name FORMAT Vertical"
            )
            logging.debug(info)
            assert False
    
        get_query = f"SELECT column1, column2, column3 FROM {dst_table_name}"
        res1 = [list(map(int, l.split())) for l in run_query(node, get_query).splitlines()]
        res2 = [
            list(map(int, l.split())) for l in run_query(node_2, get_query).splitlines()
        ]
    
        assert len(res1) + len(res2) == total_rows
    
        # Checking that all engines have made progress
        assert len(res1) > 0
>       assert len(res2) > 0
E       assert 0 > 0
E        +  where 0 = len([])
```

--- a/tests/integration/test_storage_s3_queue/test.py
+++ b/tests/integration/test_storage_s3_queue/test.py
@@ -771,7 +771,11 @@ def test_multiple_tables_streaming_sync_distributed(started_cluster, mode):
             table_name,
             mode,
             files_path,
-            additional_settings={"keeper_path": keeper_path, "s3queue_buckets": 2},
+            additional_settings={
+                "keeper_path": keeper_path,
+                "s3queue_buckets": 2,
+                **({"s3queue_processing_threads_num": 1} if mode == "ordered" else {}),
+            },
         )
 
     for instance in [node, node_2]:
@@ -805,6 +809,10 @@ def test_multiple_tables_streaming_sync_distributed(started_cluster, mode):
     res2 = [
         list(map(int, l.split())) for l in run_query(node_2, get_query).splitlines()
     ]
+
+    logging.debug(
+        f"res1 size: {len(res1)}, res2 size: {len(res2)}, total_rows: {total_rows}"
+    )
 
     assert len(res1) + len(res2) == total_rows
 

